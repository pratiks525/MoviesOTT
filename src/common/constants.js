export const IMAGES = Object.freeze({
    'poster1.jpg' : require('../assets/images/poster1.jpg'),
    'poster2.jpg' : require('../assets/images/poster2.jpg'),
    'poster3.jpg' : require('../assets/images/poster3.jpg'),
    'poster4.jpg' : require('../assets/images/poster4.jpg'),
    'poster5.jpg' : require('../assets/images/poster5.jpg'),
    'poster6.jpg' : require('../assets/images/poster6.jpg'),
    'poster7.jpg' : require('../assets/images/poster7.jpg'),
    'poster8.jpg' : require('../assets/images/poster8.jpg'),
    'poster9.jpg' : require('../assets/images/poster9.jpg'),
    missing : require('../assets/images/placeholder_for_missing_posters.png'),
    back : require('../assets/images/back.png'),
    nav_bar : require('../assets/images/nav_bar.png'),
    search : require('../assets/images/search.png'),
})

export const TITLE = 'Romantic Comedy';

export const NO_DATA_FOUND = 'No movies found';

export const MOVIES_DATA = {
    1: require('../assets/apis/CONTENTLISTINGPAGE-PAGE1.json'),
    2: require('../assets/apis/CONTENTLISTINGPAGE-PAGE2.json'),
    3: require('../assets/apis/CONTENTLISTINGPAGE-PAGE3.json')
}