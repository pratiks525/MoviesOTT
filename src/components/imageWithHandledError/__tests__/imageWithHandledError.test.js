import { render } from '@testing-library/react-native';
import React from 'react';
import ImageWithHandledError from '..';
import { IMAGES } from '../../../common/constants';

describe('ImageWithHandledError component', () => {
  test('renders image snapshot', () => {
    const container = render(<ImageWithHandledError  uri={'uri'}/>);
    expect(container).toMatchSnapshot();
  });

  test('renders missing image when provided URI', () => {
    const uri = 'uri';
    const { getByTestId } = render(<ImageWithHandledError uri={uri} />);
    const imageElement = getByTestId('thumbnail-image');
    expect(imageElement.props.source).toBe('uri');
  });

  test('renders missing image when provided URI fails to load', () => {
    const uri = undefined;
    const { getByTestId } = render(<ImageWithHandledError uri={uri} />);
    const imageElement = getByTestId('thumbnail-image');
    expect(imageElement.props.source).toBe(IMAGES.missing);
  });
});