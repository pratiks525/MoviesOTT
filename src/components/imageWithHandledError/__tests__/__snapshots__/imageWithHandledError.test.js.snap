// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageWithHandledError component renders image snapshot 1`] = `
<Image
  onError={[Function]}
  source="uri"
  testID="thumbnail-image"
/>
`;
